alacritty/src/event.rs

```diff
@@ -1551,14 +1552,15 @@ impl Processor {
         let mut scheduler = Scheduler::new(proxy.clone());
         let mut initial_window_options = Some(initial_window_options);
 
-        // NOTE: Since this takes a pointer to the winit event loop, it MUST be dropped first.
-        let mut clipboard = unsafe { Clipboard::new(event_loop.raw_display_handle()) };
-
         // Disable all device events, since we don't care about them.
         event_loop.listen_device_events(DeviceEvents::Never);
 
-        let mut initial_window_error = Ok(());
-        let result = event_loop.run(|event, event_loop| {
+        let initial_window_error = Rc::new(RefCell::new(Ok(())));
```

Comment:
Why was this changed, because of the `move`?

Reply:
yes.

Reply:
`run` moves the `event_loop` however, the closure itself could be non-movable, but we need to `move` the clipboard into the closure so it's dropped before event loop is being dropped.

Unfortunately, we can't capture just one thing by ref...

Reply:
There's no need for `Rc` or `RefCell`. Instead of cloning it you can just move a mutable reference into the event loop and that works just fine (I tested it).

Reply:
Ah, the closure is `FnMut` so we can do that. With `FnOnce` you can't iirc.